spylib/oauth/perform_token_exchange.py

```diff
@@ -0,0 +1,116 @@
+from typing import List, Union
+
+from httpx import AsyncClient, codes
+from pydantic import BaseModel, validator
+
+
+def parse_scope(scope: str) -> List[str]:
+    return scope.split(',')
+
+
+# Docstrings would be fully completed
+# Types would potentially be moved to .types
+
+
+class OfflineTokenResponse(BaseModel):   # I don't love this naming
+    """
+    [Read more about Offline access](https://shopify.dev/apps/auth/oauth/access-modes#offline-access)
+    """
+
+    access_token: str
+    """
+    An API access token that can be used to access the shop’s data as long as your app is installed. Your app should store the token somewhere to make authenticated requests for a shop’s data.
+    """
+    scope: List[str]
+    """
+    The list of access scopes that were granted to your app and are associated with the access token.
+    """
+
+    _normalize_scope = validator('scope', allow_reuse=True, pre=True)(parse_scope)
+
+
+class AssociatedUser(BaseModel):
+    id: int
+    first_name: str
+    last_name: str
+    email: str
+    email_verified: bool
+    account_owner: bool
+    locale: str
+    collaborator: bool
+
+
+class OnlineTokenResponse(BaseModel):
+    """
+    [Read more about Online access](https://shopify.dev/apps/auth/oauth/access-modes#online-access)
+    """
+
+    access_token: str
+    """
+    
+    """
+    scope: List[str]
```

Comment:
These _look_ like duplicates but would be documented differently which is why there is duplication here